src/h3lib/lib/faceijk.c

```diff
@@ -748,9 +748,14 @@ void _faceIjkToGeoBoundary(const FaceIJK* h, int res, int isPentagon,
             // find the intersection and add the lat/lon point to the result
             Vec2d inter;
             _v2dIntersect(&orig2d0, &orig2d1, edge0, edge1, &inter);
-            _hex2dToGeo(&inter, centerIJK.face, adjRes, 1,
-                        &g->verts[g->numVerts]);
-            g->numVerts++;
+            // if the intersection is the same as one of the edge endpoints,
+            // the additional vertex is on the edge itself and unnecessary
```

Comment:
Not sure if that's actually better. How about: "When a hexagon crosses over an edge, there is shape distortion similar to cutting two differently-stretched hexagons in half and then gluing them together. Most of the time this will add new vertices to the resulting 'hexagon', but if the cut goes perfectly through an existing vertex, it won't, so guard for that situation."

Reply:
This bug sounds like my bad, sorry. I should have caught that case. 

Maybe too much, but how about:

“Each face of the underlying icosahedron is a different projection plane. So if an edge of the hexagon crosses an icosahedron edge, an additional vertex must be introduced at that intersection point. Then each half of the cell edge can be projected to geographic coordinates using the appropriate icosahedron face projection. If a point of intersection occurs at a hexagon vertex, then each adjacent hexagon edge will lie completely on a single icosahedron face, and no additional vertex is requried.”

Reply:
def not too much, very clear description of whats happening 😄

Reply:
Took the recommendation, splitting into two comments, one for the outer block and one for the at-vertex check.